07_errors/nickolee/error.go

```diff
@@ -0,0 +1,21 @@
+package puppystorer
+
+import "fmt"
+
+// Error (our custom error type) wraps errors with code, message and error itself - what does this even mean??
+type Error struct {
+	Message string
+	Code    int
+}
+
+// Error() method lets us satisfy error interface for our custom error type
+func (e *Error) Error() string {
+	return fmt.Sprintf("Error code %d: %s",
```

Comment:
I would recommend output along the lines of
```
fmt.Sprintf("PuppyStoreError %d: %s", e.Code, e.Message)

// produces something like
PuppyStoreError 404: puppy not found
```
The main point is to clearly identify the error as a puppystore error in the output

Reply:
Great point. Have amended